File: get_images.py
import urllib.request
import json

places = [
    "Colosseum", "Vatican Museums", "Trevi Fountain", "Pantheon, Rome",
    "Roman Forum", "St. Peter's Basilica", "Spanish Steps", "Palatine Hill",
    "Piazza Navona", "Florence", "Venice", "Amalfi", "Positano", "Eiffel Tower",
    "Swiss Alps", "Acropolis of Athens", "Sistine Chapel"
]

results = {}

for place in places:
    try:
        url = f"https://en.wikipedia.org/w/api.php?action=query&titles={urllib.parse.quote(place)}&prop=pageimages&format=json&pithumbsize=800"
        req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
        response = urllib.request.urlopen(req)
        data = json.loads(response.read().decode())
        pages = data['query']['pages']
        for page_id in pages:
            if 'thumbnail' in pages[page_id]:
                results[place] = pages[page_id]['thumbnail']['source']
            else:
                results[place] = "NO IMAGE"
    except Exception as e:
        results[place] = str(e)

with open("urls.json", "w") as f:
    json.dump(results, f, indent=4)
print("Done")

File: get_images.php
<?php
$places = [
    "Colosseum", "Vatican Museums", "Trevi Fountain", "Pantheon, Rome",
    "Roman Forum", "St. Peter's Basilica", "Spanish Steps", "Palatine Hill",
    "Piazza Navona", "Florence", "Venice", "Amalfi", "Positano", "Eiffel Tower",
    "Swiss Alps", "Acropolis of Athens", "Sistine Chapel"
];

$results = [];
foreach ($places as $place) {
    $url = "https://en.wikipedia.org/w/api.php?action=query&titles=" . urlencode($place) . "&prop=pageimages&format=json&pithumbsize=800";
    
    $ch = curl_init();
    curl_setopt($ch, CURLOPT_URL, $url);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
    curl_setopt($ch, CURLOPT_USERAGENT, 'Mozilla/5.0');
    $output = curl_exec($ch);
    curl_close($ch);
    
    $data = json_decode($output, true);
    if (isset($data['query']['pages'])) {
        $pages = $data['query']['pages'];
        $first_page = reset($pages);
        if (isset($first_page['thumbnail']['source'])) {
            $results[$place] = $first_page['thumbnail']['source'];
        } else {
            $results[$place] = "NO IMAGE";
        }
    }
}

file_put_contents('urls.json', json_encode($results, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES));
echo "Done\n";
